### DISCUSSION TITLE
Minio Attachment not working

### DISCUSSION THREAD

**Discussion by TheOneValen** [nocodb/nocodb#4584] [Q&A] [answered]
Hi!

When I configure a minio (test successful) every attachment gets uploaded to minio. The attachment (e.g. an image file) is not visible. When deleting the attachment (clear cell) the file is still there.

Is this the expected behavior? Am I missing something (configuraiton wise)? Can I configure where the attachments are saved for each project?

Thank you!

**Comment by wingkwong** [comment]
> The attachment (e.g. an image file) is not visible.

Can you elaborate this line? Do you mean the test attachments are not found in Minio or the attachments in cells are not uploaded?

> When deleting the attachment (clear cell) the file is still there.

Currently removing attachments from the cell won't delete the file in remote sever, in this case, Minio.

**Reply by TheOneValen** [reply]
The attachment is uploaded. The browser console shows "403 Forbidden"
Will there be support for deletion in the future?

**Reply by wingkwong** [reply]
It's still not clean when you got 403. Could you record a video to show the issue? For the support for deletion, storage module will be revamped in the future.

**Reply by TheOneValen** [reply]
I just make an attachment row. Then I attach an image. Nocodb tries to show the image in the cell, it is broken. Browser says 403.

I try to make a screencast this weekend.

**Comment by ovizii** [comment, marked as answer]
@wingkwong what @TheOneValen seems to be trying to explain is exactly this issue: https://github.com/nocodb/nocodb/issues/1822 

more specifically: https://github.com/nocodb/nocodb/issues/1822#issuecomment-1114067569 

meaning uploads to S3 work but if the bucket is not public, nocodb is not able to display and download the uploaded file so basically file attachments are a one-way-street atm.

**Reply by wingkwong** [reply]
@TheOneValen Can you confirm that your bucket is private or public?

**Reply by TheOneValen** [reply]
Yes. The bucket is private. I'll watch that issue, thanks!

**Reply by TheOneValen** [reply]
Oh, will there be a fix? The issue is closed. Should I open a new one or does it not work with private buckets? Maybe that then should be documented?

**Reply by wingkwong** [reply]
The closed issue was related to another bug. Will be tracked in another issue.

**Comment by kamasylvia** [comment]
Same problem. Does anyone can answer clearly if NocoDB requires a public bucket to display or download files?

**Reply by DarkPhoenix2704** [reply]
Private Bucket support is added in https://github.com/nocodb/nocodb/releases/tag/0.252.0

By default, it will use the Presigned Url
